pro_tes/middleware/middleware_pipeline.py

```diff
@@ -0,0 +1,173 @@
+"""Set up the middleware pipeline."""
+import importlib
+import logging
+from typing import Type
+import requests
+
+from pro_tes.exceptions import InvalidMiddleware
+from pro_tes.middleware.middleware import AbstractMiddleware
+
+logger = logging.getLogger(__name__)
+
+
+class MiddlewarePipeline:
+    """
+    Middleware Pipeline.
+
+    A class for managing and processing incoming requests through a sequence of
+    middleware components.
+
+    Methods:
+        __init__(self, middlewares=None): Initializes a MiddlewarePipeline
+            instance, optionally with an initial list of middleware components.
+        add_middleware(self, middleware): Adds a middleware component to the
+            pipeline.
+        process_request(self, request, *args, **kwargs): Processes an incoming
+            request through the middleware pipeline, applying each
+            middlewares logic in sequence.
+    """
+
+    def __init__(self, middlewares=None):
+        """
+        Initialize a MiddlewarePipeline instance.
+
+        Args:
+            middlewares (list, optional): A list of middleware components to
+            initialize the pipeline. Defaults to an empty list if not provided.
```

Comment:
Second line should be indented for args and attributes, here and in other docstrings.